### ISSUE TITLE
Best practice /example request - puppet4 & multiple providers.

### ISSUE THREAD

**Issue by brettswift** [mitchellh/vagrant#7002]
This is a request for a best practice configuration of a Vagrantfile, or even just another working example to satisfy a couple of different requirements.  I have a working Vagrantfile that uses puppet4, and ties in an r10k run to collect dependencies first.   However, I can't get a working version that uses AWS as well.

I have found the Vagrantfile to be a bit klunky with Puppet4.  There's a lot of "if this then that" in the puppet file.  To use directory environments or a manifest file etc. 

This makes it complicated and klunky to configure.   When I try to add an AWS provisioner - the puppet provisioner fails.  I have to comment out the provisioner I have and do an inline provisioner.

It seems like two Vagrantfiles is the only way to do this, because I can't specify a provider specific provisioner.  I'm supposed to be able to name provisioners and then override them, so I could just set one to `echo ""` aka null, if I don't want it on a provider.

This is what I have so far for Puppet4, but adding an AWS provisioner is pretty painful.  Maybe the best thing to do right now is to have two Vagrantfiles,  but that's not very DRY.

What I have so far is documented here: 
https://github.com/brettswift/puppet-sensu_suite#vagrant-setup

**Comment by sethvargo** [comment]
Hi @brettswift 

Thank you for opening an issue. We prefer to keep the issue tracker reserved for bug reports. For best practices, please use the Vagrant mailing list or IRC channel. Thanks and sorry! :smile:

**Comment by brettswift** [comment]
Will do.  I have tried that in the past, maybe need to be more patient on the IRC channel.

I guess with this issue that I opened it was more of an "issue" with the user experience with the Puppet provisioner.  Right now I'm unable to get a working solution for puppet 4 and a combined AWS / Virtualbox provisioners in a single Vagrantfile.

I will go back to the user group and pose this question, if it can not be done I'll come back here - are feature requests valid here if this is a feature that we currently can not achieve?
